frontend/src/metabase/parameters/utils/mapping-options.ts

```diff
@@ -55,6 +58,9 @@ function buildStructuredQuerySectionOptions(
       icon: getColumnIcon(column),
       target: buildColumnTarget(query, stageIndex, column),
       isForeign: columnInfo.isFromJoin || columnInfo.isImplicitlyJoinable,
+      query,
+      stageIndex,
```

Comment:
It feels like these 2 values don't really fit into `StructuredQuerySectionOption`.
We have an example of this in [`FieldPicker.tsx`](https://github.com/metabase/metabase/blob/2739b5cf9bea0f96ce62a90f017f2daebba3b84b/frontend/src/metabase/common/components/FieldPicker/FieldPicker.tsx#L97-L100) - `item` has `column`, but `query` and `stageIndex` come from component props - not the `item`.

WDYT about making it the same here?

Reply:
That's what I tried first but there is some [custom logic](https://github.com/metabase/metabase/pull/39252/files/34ed51066b044881b1da43ebea99a7796dfd048d#diff-43ecf70f8f11873a26920b1b0a37628e34427cfaa5ed16c8fc520a832c1d65e9R162-R169) for handling models in the `getParameterMappingOptions` that I wanted to avoid duplicating.

Reply:
Fair enough :+1: